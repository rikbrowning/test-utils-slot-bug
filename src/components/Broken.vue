<template>
  <FakeComponent>
    <template v-slot:test>
      <TestComponent :msg="msg">
        <div>
          default slot of test component
        </div>
      </TestComponent>
    </template>
    <div>
      default slot of fake component
    </div>
  </FakeComponent>
</template>

<script>
import TestComponent from "./TestComponent.vue";
import FakeComponent from "./FakeComponent.vue";
export default {
  name: "Broken",
  props: {
    msg: String,
  },
  components: {
    FakeComponent,
    TestComponent,
  },
};
</script>
