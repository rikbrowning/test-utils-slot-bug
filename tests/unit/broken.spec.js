import { shallowMount } from "@vue/test-utils";
import Broken from "@/components/Broken.vue";

describe("Broken.vue", () => {
  it("renders props.msg when passed", () => {
    const msg = "new message";
    const wrapper = shallowMount(Broken, {
      propsData: { msg },
    });
    expect(wrapper).toMatchSnapshot();
  });
});
