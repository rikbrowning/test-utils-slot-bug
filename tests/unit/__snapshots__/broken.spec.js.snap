// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Broken.vue renders props.msg when passed 1`] = `
<fakecomponent-stub>
  <div>
    default slot of fake component
  </div>
</fakecomponent-stub>
`;
